test/format/wts.c

```diff
@@ -268,6 +268,41 @@ configure_debug_mode(char **p, size_t max)
     CONFIG_APPEND(*p, "]");
 }
 
+/*
+ * configure_disagg_storage --
+ *     Configure disaggregated storage settings for opening a connection.
+ */
+static void
+configure_disagg_storage(const char *home, char **p, size_t max, char *ext_cfg, size_t ext_cfg_size)
+{
+    TEST_OPTS opts;
+    char disagg_cfg[1024];
+
+    if (!g.disagg_storage_config) {
+        testutil_assert(ext_cfg_size > 0);
+        ext_cfg[0] = '\0';
+        return;
+    }
+
+    memset(&opts, 0, sizeof(opts));
+    opts.disagg_storage = true;
+
+    /*
+     * We need to cast these values. Normally, testutil allocates and fills these strings based on
+     * command line arguments and frees them when done. Format doesn't use the standard test command
+     * line parser and doesn't rely on testutil to free anything in this struct. We're only using
+     * the options struct on a temporary basis to help create the disagg configuration.
+     */
+    opts.disagg_page_log = (char *)GVS(DISAGG_PAGE_LOG);
+    opts.home = (char *)home;
+    opts.build_dir = (char *)BUILDDIR;
+    opts.palm_map_size_mb = 2048; /* 2 Gigabytes for PALM map */
```

Comment:
What does a 2GB PALM map size mean in practice? That the database can only be 2GB, or is this more like a cache size?

Reply:
Cool, thanks.